spark/v3.2/spark/src/main/java/org/apache/iceberg/spark/functions/BucketFunction.java

```diff
@@ -205,7 +211,6 @@ public static Integer invoke(int numBuckets, UTF8String value) {
         return null;
       }
 
-      // TODO - We can probably hash the bytes directly given they're already UTF-8 input.
```

Comment:
I'd say it is a valid TODO item. We actually have a similar optimization internally.

Reply:
You're right. We left the optimization in truncate but didn't move forward with it for `bucket`. My mistake.

Reply:
Yeah, we should be able to pass the backing byte array. Let's do this in a follow-up for 3.3.